External/SocketRocket/SPRWebSocket.m

```diff
@@ -465,10 +465,10 @@ - (void)_readHTTPHeader;
     }
                         
     [self _readUntilHeaderCompleteWithCallback:^(SPRWebSocket *self,  NSData *data) {
-        CFHTTPMessageAppendBytes(_receivedHTTPHeaders, (const UInt8 *)data.bytes, data.length);
+        CFHTTPMessageAppendBytes(self->_receivedHTTPHeaders, (const UInt8 *)data.bytes, data.length);
```

Comment:
**DISCLAIMER:** I've switched to Swift, 24x7, few years ago, so my most sincere apologies if i literally forgot all the things!!

1. `_receivedHTTPHeaders` == `self.receivedHTTPHeaders`
2. Having `self` inside the callback block should be causing `self` to get retained
3. If `self` is retained, `receivedHTTPHeaders` should be retained as well?

How about something in the lines of...

```
__weak typeof(self) wself = self;
[self _readUntilHeaderCompleteWithCallback:^(SPRWebSocket *self,  NSData *data) {
    [wself handleHeaderCallback: ...];
}];
```

Thoughts?

Reply:
This one is a little tricky to follow because of the shadowing of `self` by the block parameter. The difference is that `_receivedHTTPHeaders` by itself implicitly uses the `self` from the method defining the block and is captured, but explicit references to `self` within the block use the block's `self` parameter, which is not captured. This is the same pattern used in `_handleFrameHeader:curData:` and `_readFrameContinue` to avoid retain cycles there, they just missed this one. Later versions of SocketRocket renamed these block parameters to `sself` to make it slightly clearer that these are different.

I figured it was best to be consistent with the existing code but we can certainly do something different here if you prefer. I did test this before submitting to confirm that it breaks the retain cycle and fixes the leak I was seeing.

Reply:
AHHHHH i totally missed the `self` parameter, thanks for bringing that up!!! Everything makes sense now (thank you!!)

Do you happen to have any special steps to trigger this glitch? (in which state was the lib while you lost connectivity? was it authenticated? downloading a huge batch of changes?)

Thanks in advance Matt!

Reply:
The easiest way I can think of to reproduce this is to change SPWebsocketURL to point at a server with an untrusted server certificate.

In my case I'm authenticated and successfully connected, then I VPN into this network that performs HTTPS intercept, the websocket connection drops, then keeps trying to reestablish and failing due to the untrusted server certificate from the proxy performing intercept.